### PR TITLE
Scheduler tests

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Adding Scheduler tests some initial integration tests for the new post pone & skip functionality.

### What issues does this PR fix or reference?
N/A

### Previous Behavior
Tests did not exist.

### New Behavior
Adding integration tests.

### Tests written?
Yes

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by thatch45 — CHANGES_REQUESTED** [review]
That docstring is the only thing I see so far.

File: salt/utils/schedule.py [hunk 1]
```diff
@@ -830,6 +830,14 @@ def get_next_fire_time(self, name):
         evt.fire_event({'complete': True, 'next_fire_time': _next_fire_time},
                        tag='/salt/minion/minion_schedule_next_fire_time_complete')
 
+    def job_status(self, name):
+        '''
+        Disable a job in the scheduler. Ignores jobs from pillar
```
[the remark this answers is not in the document]
Comment (garethgreenaway) [reply salt/utils/schedule.py]: Yes.  Yes we do.

**Review by thatch45 — APPROVED** [review]

**Comment by garethgreenaway** [comment]
re-run ubuntu

**Comment by rallytime** [comment]
re-run py2

**Comment by rallytime** [comment]
re-run centos

**Comment by rallytime** [comment]
Heh. I was just testing the new command trigger for the centos tests. But it looks like it didn't work. The test failures on centos are not related to this change.

**Review by cachedout — CHANGES_REQUESTED** [review]
This is not the correct location for new tests. Note that they did not get picked up by the test runner. I suggest they get moved to a new directory inside the integration test suite. Either @rallytime or @Ch3LL can help you with details here.

**Comment by rallytime** [comment]
Ah, good catch @cachedout. I just checked and the new changes are running the tests. Thanks for updating this @garethgreenaway.

**Review by rallytime — APPROVED** [review]

**Comment by garethgreenaway** [comment]
re-run centos

**Comment by rallytime** [comment]
Go Go Jenkins!

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by garethgreenaway** [comment]
```
 ✘ gareth@blackoil  ~/code/salt/salt/utils   scheduler_tests  sudo python /home/gareth/code/salt/tests/runtests.py --scheduler 
[sudo] password for gareth: 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Python Version: 2.7.14 (default, Sep 17 2017, 18:50:44) [GCC 7.2.0]
 * Transplanting configuration files to '/tmp/salt-tests-tmpdir/config'
 * Current Directory: /home/gareth/code/salt
 * Test suite is running under PID 19817
 * Logging tests on /tmp/salt-runtests.log
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Max open files settings is too low (soft: 1024, hard: 4096) for running the tests
 * Trying to raise the limits to soft: 3072, hard: 4096
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Setting up Salt daemons to execute tests
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Starting salt-master ... STARTED!                                                                                                                                                                                                                                            
 * Starting salt-minion ... STARTED!                                                                                                                                                                                                                                            
 * Starting sub salt-minion ... STARTED!                                                                                                                                                                                                                                        
 * Starting syndic salt-master ... STARTED!                                                                                                                                                                                                                                     
 * Starting salt-syndic ... STARTED!                                                                                                                                                                                                                                            
================================================================================================================================================================================================================================================================================
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Starting Scheduler Tests
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.....
----------------------------------------------------------------------
Ran 5 tests in 2.746s

OK

============================================================================================================================  Overall Tests Report  ============================================================================================================================
***  No Problems Found While Running Tests  ************************************************************************************************************************************************************************************************************************************
================================================================================================================================================================================================================================================================================
OK (total=5, skipped=0, passed=5, failures=0, errors=0) 
============================================================================================================================  Overall Tests Report  ============================================================================================================================
 gareth@blackoil  ~/code/salt/salt/utils   scheduler_tests  
```

**Comment by garethgreenaway** [comment]
```
 gareth@blackoil  ~/code/salt   scheduler_tests  sudo python3 /home/gareth/code/salt/tests/runtests.py --scheduler
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Python Version: 3.6.3 (default, Oct 3 2017, 21:16:13) [GCC 7.2.0]
 * Transplanting configuration files to '/tmp/salt-tests-tmpdir/config'
 * Current Directory: /home/gareth/code/salt
 * Test suite is running under PID 24195
 * Logging tests on /tmp/salt-runtests.log
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Max open files settings is too low (soft: 1024, hard: 4096) for running the tests
 * Trying to raise the limits to soft: 3072, hard: 4096
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Setting up Salt daemons to execute tests
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * Starting salt-master ... STARTED!                                                                                                                                                                                                                                            
 * Starting salt-minion ... STARTED!                                                                                                                                                                                                                                            
 * Starting sub salt-minion ... STARTED!                                                                                                                                                                                                                                        
 * Starting syndic salt-master ... STARTED!                                                                                                                                                                                                                                     
 * Starting salt-syndic ... STARTED!                                                                                                                                                                                                                                            
================================================================================================================================================================================================================================================================================
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Starting Scheduler Tests
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.....
----------------------------------------------------------------------
Ran 5 tests in 2.164s

OK

============================================================================================================================  Overall Tests Report  ============================================================================================================================
***  No Problems Found While Running Tests  ************************************************************************************************************************************************************************************************************************************
================================================================================================================================================================================================================================================================================
OK (total=5, skipped=0, passed=5, failures=0, errors=0) 
============================================================================================================================  Overall Tests Report  ============================================================================================================================
 gareth@blackoil  ~/code/salt   scheduler_tests  
```

**Pull request merged**